src/Pmi/Controller/BiobankController.php

```diff
@@ -225,13 +229,33 @@ public function orderAction($biobankId, $orderId, Application $app, Request $req
                         if ($finalizeForm["finalized_samples"]->getData() && is_array($finalizeForm["finalized_samples"]->getData())) {
                             $samples = array_values($finalizeForm["finalized_samples"]->getData());
                         }
+                        // Check centrifuge type for dv kit orders
+                        if ($order->get('type') === 'kit' && empty($order->get('processed_centrifuge_type'))) {
+                            if ($finalizeForm->has('processed_centrifuge_type')) {
+                                $centrifugeType = $finalizeForm['processed_centrifuge_type']->getData();
+                            } elseif (!empty($site['centrifuge_type'])) {
+                                $centrifugeType = $site['centrifuge_type'];
+                            }
+                        }
+                        if ($order->get('type') === 'kit' && empty($order->get('processed_centrifuge_type'))) {
+                            $site = $app['em']->getRepository('sites')->fetchOneBy([
+                                'deleted' => 0,
+                                'google_group' => $app->getSiteId()
+                            ]);
+                            if (!empty($site['centrifuge_type'])) {
+                                $order->set('processed_centrifuge_type', $site['centrifuge_type']);
+                            }
+                        }
```

Comment:
1. This if clause is the same as above. Not sure why it needs to be in a separate block.
2. Don't you already have the site retrieved from the database from above?

Reply:
Sorry, forgot to remove some of the unwanted code that I have written previously.